src/app/core/services/cache-repo-from-url.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CacheRepoFromUrlService {
```

Comment:
This has a very similar name to `RepoUrlCacheService`, maybe we could rename it to `RepoSessionStorageService`? This allows for lesser confusion esp within the SessionSelectionComponent and is a more direct approach, indicating it's about storing a repo within session storage.